certbot-dns-google/certbot_dns_google/_internal/dns_google.py

```diff
@@ -118,10 +118,10 @@ def add_txt_record(self, domain, record_name, record_content, record_ttl):
 
         record_contents = self.get_existing_txt_rrset(zone_id, record_name)
         if record_contents is None:
```

Comment:
I'm happy to add this. This is good for consumers who are willing to jump into this code and understand what it does. For those who don't, when we run into the situation:
> If there are actually records here, this will fail with HTTP 409/412 API errors.

how explicit/informative are we with our users when we throw the exception further down ? (line 169)

https://github.com/certbot/certbot/blob/82fc034715da37160d87666ccc50962a12d89796/certbot-dns-google/certbot_dns_google/_internal/dns_google.py#L163-L170

if we you just throw a HTTP 412, people would be puzzled and would need to go and study the Google DNS API (which poorly documents this error btw) and then figured out that the actual problem was that `certbot` was missing `.list` permission ! . 

In that case, I wonder if we should rather fail fast ? That would be material for a different PR though, but since you brought this up I felt relevant to expand the discussion.

Reply:
For me, the comment serves only to contextualize the code for other developers.

If we want to say that the errors should be easier for end-users to understand and overcome, then I agree:

- We could improve the message emitted in `get_existing_txt_rrset`, to not be wildcard-specific, and also to provide a suggested remedial action (create a set of credentials with full DNS permissions).
- We could add the same caveat to the docs at https://certbot-dns-google.readthedocs.io/en/stable/.

However, as far as I can tell, nobody has complained about this permission thing ... ever. I don't think we should spend effort on improving the UX of it if it's purely a hypothetical problem.

Reply:
Agree 👍 . My suggestion was coming basically from my own experience when I faced the bug that we fixed on #8549 and #8521. Let's keep an eye on future complaints if they come and we'll revisit this if necessary 😄 .